e2e/e2e_test.go

```diff
@@ -236,6 +480,44 @@ func podObject(label, annotations map[string]string) *core.Pod {
 	}
 }
 
+func replicaSetObject(replicaCount int32, label map[string]string, annotations map[string]string) *apps.ReplicaSet {
+	numReplicas := &replicaCount
+
+	return &apps.ReplicaSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ReplicaSet",
+		},
```

Comment:
nit: I think this is not needed. The kubernetes API is smart enough to know that an `apps.ReplicaSet` object has kind `ReplicaSet` . 

You can keep it if you want.

Reply:
I'll remove it, but I won't resolve this conversation until I check that nothing breaks, just to be safe.